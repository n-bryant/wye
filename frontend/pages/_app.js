import React from "react";
import App from "next/app";

import Page from "../src/components/Page";
import { ApolloProvider } from "react-apollo";
import withData from "../lib/withData";

import CssBaseline from "@material-ui/core/CssBaseline";

import NextNProgress from "../src/components/NextNProgress";

// set up a context provider
const AppContext = React.createContext({});
export const AppContextProvider = AppContext.Provider;
export const AppContextConsumer = AppContext.Consumer;

// reducer action and content options
export const ACTIONS = {
  SET_USERS: "setUsers",
  SET_CONTENT: "setContent"
};
export const CONTENT_OPTIONS = {
  WELCOME: "WELCOME",
  FORM: "FORM",
  RECOMMENDATIONS: "RECOMMENDATIONS"
};

// initial state values
export const INITIAL_STATE = {
  content: CONTENT_OPTIONS.WELCOME,
  users: []
};

/**
 * reducer for updating state values
 * @param {Object} state
 * @param {Object} action
 * @returns {Object}
 */
export const reducer = (state, action) => {
  switch (action.type) {
    case ACTIONS.SET_USERS:
      return {
        ...state,
        users: action.value
      };
    case ACTIONS.SET_CONTENT:
      return {
        ...state,
        content: action.value
      };
    default:
      throw new Error();
  }
};

export class WyeApp extends App {
  componentDidMount() {
    // Remove the server-side injected CSS.
    const jssStyles = document.querySelector("#jss-server-side");
    if (jssStyles) {
      jssStyles.parentElement.removeChild(jssStyles);
    }
  }

  static async getInitialProps({ Component, ctx }) {
    let pageProps = {};
    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps(ctx);
    }
    // expose query to the user
    pageProps.query = ctx.query;
    return { pageProps };
  }

  render() {
    return <WyeWithState {...this.props} />;
  }
}

// app with state
export const WyeWithState = props => {
  const { Component, apollo, pageProps } = props;
  const [state, dispatch] = React.useReducer(reducer, INITIAL_STATE);

  return (
    <React.Fragment>
      <CssBaseline />
      <NextNProgress />
      <ApolloProvider client={apollo}>
        <AppContextProvider
          value={{
            state,
            dispatch
          }}
        >
          <Page>
            <Component {...pageProps} />
          </Page>
        </AppContextProvider>
      </ApolloProvider>
    </React.Fragment>
  );
};

// app with apollo client
export const WyeWithData = withData(WyeApp);

export default WyeWithData;
