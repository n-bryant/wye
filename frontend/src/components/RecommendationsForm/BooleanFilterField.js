import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Switch from "@material-ui/core/Switch";
import Typography from "@material-ui/core/Typography";

import styles from "./BooleanFilterField.styles";

/**
 * renders a field for electing a boolean value in the form of a Switch
 */
export const BooleanFilterField = props => {
  const classnames = BooleanFilterField.classnames(props);
  const { classes, name, type, label, setFieldValue, value } = props;
  const [checked, setChecked] = React.useState(false);

  /**
   * handle changes to the Switch,
   * and update formik field value
   * @param {String} option
   */
  const handleChange = event => {
    setChecked(event.target.checked);
    setFieldValue(name, {
      ...value,
      gameFilters: {
        ...value.gameFilters,
        [type]: event.target.checked
      }
    });
  };

  return (
    <div className={classnames.root()}>
      <Typography
        className={classnames.element("label")}
        variant="body2"
        gutterBottom={true}
      >
        {label}
      </Typography>
      <div>
        <Switch
          classes={{
            switchBase: classes.switch,
            checked: classes.switchChecked,
            track: classes.switch
          }}
          checked={checked}
          onChange={handleChange}
          value={label}
          color="primary"
          inputProps={{ "aria-label": `${type}-checkbox` }}
        />
      </div>
    </div>
  );
};
BooleanFilterField.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}BooleanFilterField`
);
BooleanFilterField.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    label: PropTypes.string,
    switch: PropTypes.string,
    switchChecked: PropTypes.string
  }),
  // Formik field name
  name: PropTypes.string,
  // the label to display for the field
  label: PropTypes.string,
  // the type of field
  type: PropTypes.oneOf(["freeToPlay", "onSale"]).isRequired,
  // gameFilters field value from Formik form
  value: PropTypes.object,
  // handler for updating users field value
  setFieldValue: PropTypes.func
};
BooleanFilterField.defaultProps = {
  classes: {}
};

// apply styles
export default withStyles(styles)(BooleanFilterField);
