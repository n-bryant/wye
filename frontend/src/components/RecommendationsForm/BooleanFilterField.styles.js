const styles = theme => ({
  root: {},
  label: {
    color: theme.palette.common.blue6
  },
  switch: {
    color: theme.palette.common.white
  },
  switchChecked: {
    color: `${theme.palette.common.blue6} !important`
  }
});
export default styles;
