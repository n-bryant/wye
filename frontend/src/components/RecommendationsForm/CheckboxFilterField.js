import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Typography from "@material-ui/core/Typography";

import styles from "./CheckboxFilterField.styles";

/**
 * renders a field for electing a selection of values in the form of a checkbox list
 */
export const CheckboxFilterField = props => {
  const classnames = CheckboxFilterField.classnames(props);
  const {
    classes,
    name,
    category,
    type,
    label,
    setFieldValue,
    value,
    options
  } = props;

  // set up initial state values
  const perPage = 20;
  let initialValuesState = {};
  options.forEach(option => {
    initialValuesState[option] = false;
  });
  const [checkedOptions, setCheckedOptions] = React.useState(
    initialValuesState
  );
  const [currentPage, setCurrentPage] = React.useState(1);

  /**
   * handle changes to the CheckBox options,
   * and update formik field value
   * @param {String} option
   */
  const handleChange = option => event => {
    const newValue = { ...checkedOptions, [option]: event.target.checked };
    setCheckedOptions(newValue);
    let valueToSet = {
      ...value,
      [category]: {
        ...value[category],
        [type]: Object.keys(newValue).filter(key => Boolean(newValue[key]))
      }
    };
    if (Object.keys(newValue).every(key => !newValue[key])) {
      delete valueToSet[category][type];
    }
    setFieldValue(name, valueToSet);
  };

  // handle scroll event and loading of more items
  const handleScroll = event => {
    const containerHeight = event.target.getBoundingClientRect().height;
    const optionsListHeight = event.target.firstElementChild.getBoundingClientRect()
      .height;
    if (event.target.scrollTop >= optionsListHeight - containerHeight - 84) {
      setCurrentPage(currentPage + 1);
    }
  };

  const optionsCount = Object.keys(checkedOptions).length;
  const selectedOptionsCount = Object.keys(checkedOptions).filter(
    key => !!checkedOptions[key]
  ).length;
  return (
    <div className={classnames.root()}>
      <Typography
        className={classnames.element("label")}
        variant="body2"
        gutterBottom={true}
      >
        {`${label} ${optionsCount} ${
          selectedOptionsCount > 0 ? `- (${selectedOptionsCount} selected)` : ""
        }`}
      </Typography>
      <div
        className={classnames.element("optionsContainer")}
        onScroll={handleScroll}
      >
        <div className={classnames.element("options")}>
          {options.slice(0, currentPage * perPage).map((option, index) => (
            <FormControlLabel
              key={index}
              control={
                <Checkbox
                  classes={{
                    root: classes.checkbox,
                    checked: classes.checkboxChecked
                  }}
                  checked={checkedOptions[option]}
                  color="default"
                  onChange={handleChange(option)}
                  value={option}
                  inputProps={{ "aria-label": `${name}-option_${option}` }}
                />
              }
              label={option}
            />
          ))}
        </div>
      </div>
    </div>
  );
};
CheckboxFilterField.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}CheckboxFilterField`
);
CheckboxFilterField.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    checkbox: PropTypes.string,
    label: PropTypes.string,
    optionsContainer: PropTypes.string,
    checkboxChecked: PropTypes.string
  }),
  // Formik field name
  name: PropTypes.string,
  // which filter category is the field for
  category: PropTypes.oneOf(["gameFilters", "playerFilters"]),
  // which filter type is it
  type: PropTypes.oneOf([
    "ownedBy",
    "recentlyPlayedBy",
    "publishers_in",
    "developers_in",
    "genres_in",
    "tags_in"
  ]).isRequired,
  // the label to display for the field
  label: PropTypes.string,
  // playerFilters field value from Formik form
  value: PropTypes.object,
  // handler for updating users field value
  setFieldValue: PropTypes.func,
  // the available steamids to select
  options: PropTypes.array
};
CheckboxFilterField.defaultProps = {
  classes: {}
};

// apply styles
export default withStyles(styles)(CheckboxFilterField);
