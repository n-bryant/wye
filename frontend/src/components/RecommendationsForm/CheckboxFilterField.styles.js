const styles = theme => ({
  root: {},
  checkbox: {
    color: theme.palette.common.white
  },
  label: {
    color: theme.palette.common.blue6
  },
  optionsContainer: {
    display: "flex",
    flexDirection: "column",
    maxHeight: "250px",
    overflow: "auto"
  },
  checkboxChecked: {
    color: theme.palette.common.blue6
  }
});
export default styles;
