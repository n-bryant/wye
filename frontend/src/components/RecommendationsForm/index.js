import React from "react";
import PropTypes from "prop-types";

import get from "lodash.get";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Icon from "@mdi/react";
import { mdiClose } from "@mdi/js";
import { Typography } from "@material-ui/core";

import {
  AppContextConsumer,
  ACTIONS,
  CONTENT_OPTIONS
} from "../../../pages/index";
import UsersField from "./UsersField";
import ActionButton from "../ActionButton";
import styles from "./index.styles";

/**
 * renders instructions for how to start getting recommendations and an input for user IDs
 */
export const RecommendationsForm = props => {
  const classnames = RecommendationsForm.classnames(props);
  const { dispatch, users } = props;
  const [usersValue, setUsersValue] = React.useState(users);

  // disable the submit button if no SteamIDs have been added
  const submitDisabled = !usersValue.length > 0;

  return (
    <div className={classnames.root()}>
      <Icon
        className={classnames.element("closeButton")}
        path={mdiClose}
        onClick={() =>
          dispatch({
            type: ACTIONS.SET_CONTENT,
            value: CONTENT_OPTIONS.WELCOME
          })
        }
      />
      <div className={classnames.element("formInstructions")}>
        <Typography className={classnames.element("title")} variant="h1">
          Getting Your Recommendations:
        </Typography>
        <Typography variant="body1">
          Wye will compare the libraries of each provided SteamID to generate
          your recommendations. By default, Wye will recommend the top rated
          multiplayer game owned by the majority of the entered SteamIDs, but
          additional filters will become available after the initial search. Use
          the field below to add up to six SteamIDs, then click "Go!".
        </Typography>
      </div>
      <div className={classnames.element("userFieldContainer")}>
        <UsersField usersValue={usersValue} setUsersValue={setUsersValue} />
      </div>
      <ActionButton
        className={classnames.element("submitButton")}
        disabled={submitDisabled}
        onClick={() => {
          dispatch({ type: ACTIONS.SET_USERS, value: usersValue });
          dispatch({
            type: ACTIONS.SET_CONTENT,
            value: CONTENT_OPTIONS.RECOMMENDATIONS
          });
        }}
      />
    </div>
  );
};
RecommendationsForm.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}RecommendationsForm`
);
RecommendationsForm.propTypes = {
  // styles to apply to the form
  classes: PropTypes.shape({
    root: PropTypes.string,
    closeButton: PropTypes.string,
    formInstructions: PropTypes.string,
    title: PropTypes.string,
    userFieldContainer: PropTypes.string,
    submitButton: PropTypes.string
  }),
  // dispatch handler for updating state
  dispatch: PropTypes.func,
  // users value from context
  users: PropTypes.array
};
RecommendationsForm.defaultProps = {
  classes: {}
};

// Apply styles to the RecommendationsForm
export const StyledRecommendationsForm = withStyles(styles)(
  RecommendationsForm
);

/**
 * Renders a StyledRecommendationsForm with context from the index page
 */
export const RecommendationsFormWithContext = props => (
  <AppContextConsumer>
    {context => {
      const dispatch = get(context, "dispatch", () => {});
      const users = get(context, ["state", "users"], []);
      return (
        <StyledRecommendationsForm
          dispatch={dispatch}
          users={users}
          {...props}
        />
      );
    }}
  </AppContextConsumer>
);

export default RecommendationsFormWithContext;
