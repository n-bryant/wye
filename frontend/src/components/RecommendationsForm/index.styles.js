const styles = theme => ({
  root: {
    height: "100%",
    width: "100%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    position: "relative",
    paddingTop: theme.spacing(6)
  },
  closeButton: {
    fill: "hsla(0,0%,100%,0.2)",
    cursor: "pointer",
    position: "absolute",
    top: theme.spacing(1),
    right: theme.spacing(1),
    height: "1.5rem",
    transition: ".25s ease-in-out",
    "&:hover, &:active": {
      fill: theme.palette.common.white,
      transition: ".25s ease-in-out"
    }
  },
  formInstructions: {
    marginBottom: theme.spacing(4),
    width: "75%",
    textAlign: "center",
    [theme.breakpoints.down("sm")]: {
      width: "100%",
      padding: theme.spacing(1)
    }
  },
  title: {
    marginBottom: theme.spacing(2),
    color: theme.palette.common.blue6
  },
  userFieldContainer: {
    width: "100%",
    display: "flex",
    justifyContent: "center"
  },
  submitButton: {}
});
export default styles;
