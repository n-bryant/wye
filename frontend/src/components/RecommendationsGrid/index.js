import React from "react";
import PropTypes from "prop-types";

import get from "lodash.get";
import gql from "graphql-tag";
import { useQuery } from "@apollo/react-hooks";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Icon from "@mdi/react";
import { mdiPlaylistEdit, mdiPlaylistPlus } from "@mdi/js";

import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

import {
  AppContextConsumer,
  ACTIONS,
  CONTENT_OPTIONS
} from "../../../pages/index";
import LoadingState from "./LoadingState";
import UserAvatar from "./UserAvatar";
import Item from "./Item";
import styles from "./index.styles";

// recommendations query
export const GET_RECOMMENDATIONS_QUERY = gql`
  query GET_RECOMMENDATIONS_QUERY($users: [ID!]!) {
    recommendations(users: $users) {
      pageInfo {
        totalCount
      }
      userDetails {
        id
        profileUrl
        avatarName
        avatarImageUrl
        onlineStatus
        lastOnlineTime
      }
      edges {
        node {
          game {
            appid
            name
            developers
            publishers
            genres
            tags
            freeToPlay
            onSale
            discount
            initialPrice
            finalPrice
            userRating
            logoImageUrl
            heroImageUrl
          }
          ownedBy
          recentlyPlayedBy
          playtime {
            id
            hoursPlayed
          }
        }
      }
    }
  }
`;

/**
 * renders a grid of recommendations for the provided users
 */
export const RecommendationsGrid = props => {
  const classnames = RecommendationsGrid.classnames(props);
  const { users, dispatch } = props;

  // with query
  const { loading, error, data } = useQuery(GET_RECOMMENDATIONS_QUERY, {
    variables: { users }
  });
  // console.log(data);
  const edges = get(data, ["recommendations", "edges"], []);
  const userDetails = get(data, ["recommendations", "userDetails"], []);

  // loading state
  if (loading) {
    return <LoadingState />;
  }

  // error state
  if (error) {
    return <div>error placeholder</div>;
  }

  return (
    <Grid className={classnames.root()} container spacing={2}>
      <Grid
        className={classnames.element("subGrid", {
          withDivider: true
        })}
        container
        justify="center"
        spacing={3}
      >
        <Grid item xs={12}>
          <Typography
            className={classnames.element("subGridTitle")}
            variant="h1"
          >
            Showing Recommendations For:
          </Typography>
        </Grid>
        <Grid
          className={classnames.element("avatarContainer")}
          container
          item
          xs={12}
          spacing={2}
        >
          {userDetails.map((data, index) => (
            <UserAvatar key={index} data={data} />
          ))}
        </Grid>
        <Grid container item justify={"center"} xs={12} md={8} spacing={2}>
          <Grid item xs={6}>
            <Button
              className={classnames.element("editButton")}
              onClick={() => {
                dispatch({
                  type: ACTIONS.SET_CONTENT,
                  value: CONTENT_OPTIONS.FORM
                });
              }}
            >
              <Icon
                className={classnames.element("editIcon")}
                path={mdiPlaylistEdit}
              />
              <Typography variant="body2">Edit SteamIDs</Typography>
            </Button>
          </Grid>
          <Grid item xs={6}>
            <Button className={classnames.element("editButton")}>
              <Icon
                className={classnames.element("editIcon")}
                path={mdiPlaylistPlus}
              />
              <Typography variant="body2">Additional Filters</Typography>
            </Button>
          </Grid>
        </Grid>
      </Grid>
      <Grid
        className={classnames.element("subGrid")}
        container
        justify="center"
        spacing={3}
      >
        <Grid item xs={12}>
          <Typography
            className={classnames.element("subGridTitle", {
              centeredWhenLg: true
            })}
            variant="h3"
          >
            Top Recommendation:
          </Typography>
        </Grid>
        <Item featured={true} userDetails={userDetails} data={edges[0].node} />
      </Grid>
      <Grid
        className={classnames.element("subGrid")}
        justify="center"
        container
        spacing={3}
      >
        <Grid item xs={12}>
          <Typography
            className={classnames.element("subGridTitle")}
            variant="h3"
          >
            Other Recommendations:
          </Typography>
        </Grid>
        {edges.slice(1, 11).map((item, index) => (
          <Item
            key={index}
            featured={false}
            userDetails={userDetails}
            data={item.node}
          />
        ))}
      </Grid>
    </Grid>
  );
};
RecommendationsGrid.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}RecommendationsGrid`
);
RecommendationsGrid.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    subGrid: PropTypes.string,
    editButton: PropTypes.string,
    editIcon: PropTypes.string,
    subGridWithDivider: PropTypes.string,
    avatarContainer: PropTypes.string,
    subGridTitle: PropTypes.string,
    subGridTitleCenteredWhenLg: PropTypes.string
  }),
  // users to get recommendations for
  users: PropTypes.array,
  // handler for updating app level state
  dispatch: PropTypes.func
};
RecommendationsGrid.defaultProps = {
  classes: {},
  users: [],
  dispatch: () => {}
};

// apply styles
export const StyledRecommendationsGrid = withStyles(styles)(
  RecommendationsGrid
);

/**
 * Renders a StyledRecommendationsGrid with context from the index page
 */
export const RecommendationsGridWithContext = props => (
  <AppContextConsumer>
    {context => {
      const users = get(context, ["state", "users"], []);
      const dispatch = get(context, "dispatch", () => {});
      return (
        <StyledRecommendationsGrid
          users={users}
          dispatch={dispatch}
          {...props}
        />
      );
    }}
  </AppContextConsumer>
);

export default RecommendationsGridWithContext;
