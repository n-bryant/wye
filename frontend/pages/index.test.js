import { CONTENT_OPTIONS, ACTIONS } from "./_app";
import { Index } from "./index";
import RecommendationsForm from "../src/components/RecommendationsForm";
import RecommendationsGrid from "../src/components/RecommendationsGrid";

describe("index", () => {
  const props = {
    classes: {
      root: "root",
      container: "container",
      form: "form",
      main: "main",
      contentContainer: "contentContainer",
      content: "content"
    },
    content: CONTENT_OPTIONS.WELCOME
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render successfully", () => {
    const wrapper = shallow(<Index {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render the recommendations form if the content state value is set to the form content option", () => {
    const propsWithFormContent = {
      ...props,
      content: CONTENT_OPTIONS.FORM
    };
    const wrapper = shallow(<Index {...propsWithFormContent} />);
    expect(wrapper.find(RecommendationsForm).length).toBe(1);
  });

  it("should render the recommendations grid if the content state value is set to the recommendations content option", () => {
    const propsWithFormContent = {
      ...props,
      content: CONTENT_OPTIONS.RECOMMENDATIONS
    };
    const wrapper = shallow(<Index {...propsWithFormContent} />);
    expect(wrapper.find(RecommendationsGrid).length).toBe(1);
  });
});
