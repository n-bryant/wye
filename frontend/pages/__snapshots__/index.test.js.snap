// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render successfully 1`] = `
<div
  className="wye-Index root"
>
  <WithStyles(ForwardRef(Container))
    className="wye-Index__container container"
    disableGutters={true}
    maxWidth={false}
  >
    <div
      className="wye-Index__main main"
    >
      <div
        className="wye-Index__contentContainer contentContainer"
      >
        <div
          className="wye-Index__content content"
        >
          <WelcomeEmptyStateWithContext />
        </div>
      </div>
    </div>
  </WithStyles(ForwardRef(Container))>
  <WithStyles(IndexPageBackground)
    classes={Object {}}
  />
</div>
`;
